src/ipld.rs

```diff
@@ -11,7 +12,24 @@ use core::fmt;
 
 use cid::Cid;
 
-use crate::error::TypeError;
+/// Error when accessing IPLD List or Map elements.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum AccessError {
+    /// Error message describing the error.
+    Message(String),
```

Comment:
I was actually hoping to enumerate the cases rather than have a generic "Message". Is that possible, or just too much work.

Personally, I'm not a huge fan of catch-all "message"/"custom"/"other" error variants (although I understand that they're sometimes required).

Reply:
I'll look if I find any structure. I'm more of a fan of such error messages as they can really be specific to what's going on. And in this case (accessing some field) it's really kind of a single error case, with a nice description.

Reply:
I agree we need something descriptive, but I prefer being able to programatically handle errors where possible. Of course, when we're dealing with malformed data... maybe this doesn't matter as much.

Reply:
Your message and my new push just crossed :)